README.md

```diff
@@ -1,21 +1,161 @@
 
-## Contribute
+P2P Editor is a code editor that works in the browser which lets you share live coding sessions. See it by yourself sharing a new session:
 
-TODO
+1. [Open a P2P Editor session in your browser](http://p2p-editor.codely.tv)
+2. Share your session URL with someone else (or even with yourself in another browser window 😬)
+3. Start typing and enjoy the real time experience!
 
-## License
+The functionalities are limited for now, but we have a [roadmap](https://github.com/CodelyTV/p2p-editor/wiki) and some [good first issues](https://github.com/CodelyTV/p2p-editor/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22good+first+issue%22) to make P2P Editor awesome. If you want to get involved take a look of [Contributing](#-contributing) section. 
 
-MIT
+## 🤔 Why P2P?
+
+We have removed the need of having a centralized server where all the changes get saved. The communication is done from one peer (a connected user) to the other one. By doing this we have some benefits, but we also have to take into account the added complexity:
+
+### Benefits of P2P
+
+- Direct communication: reduce latency and bandwidth.
+- No servers: no cloud, no scaling, no deploy, no devops...
+- No SaaS:
+  - Availability.
+  - No usage restriction.
+  - No unknown (malicious or not) usage of your data.
+
+### Added complexities of being fully distributed
+
+- Data consistency.
+- Network partition / topology.
+- Peer discovery and connection establishment.
+- Authentication / Authorization.
+- Security and privacy.
+
+We have a particular interest in P2P application development. You can read more in the [About](#-about) section.
+
+## 🏗️ Architecture
+
+![Architecture Diagram](assets/architecture_diagram.jpg)
+
+P2P Editor building blocks:
+- Code editor: [Ace](https://github.com/ajaxorg/ace)
+- Database: append only log ([hypercore](https://github.com/mafintosh/hypercore))
+- Communication: WebRTC RTCDataChannel ([webrtc-swarm](https://github.com/mafintosh/webrtc-swarm))
+- Peer discovery: WebRTC signaling server ([signalhub](https://github.com/mafintosh/signalhub))
+- Storage: RAM
+
+Workflow:
+
+1. Peers are discovered via signaling server.
+2. For each change made in the editor, a delta representing that change is appended to hypercore database.
+3. Hypercore database is constantly replicating between all peers via WebRTC RTCDataChannel.
+4. When a peer receives a delta it is applied to its editor instance.
+
+## 🚀 Environment setup
+
+### 🐳 Needed tools
+
+1. [Install Docker](https://www.docker.com/get-started) 
+2. Clone this project: `git clone https://github.com/CodelyTV/p2p-editor`
+3. Move to the project folder: `cd p2p-editor`
+
+### 🛠️ Environment configuration
+
+1. Copy the default environment variables: `cp .env.dist .env`
+2. Modify the environment variables if needed: `vim .env`
+
+### 🌍 Application execution
+
+1. Start the application with Docker Compose: `docker-compose up`
+2. Go to [http://localhost:8000](http://localhost:8000)
```

Comment:
Didn't we changed the default port to something more random to avoid collisions with common local development environments?

Suggested port: `3617` (the closest numbers to the `edit` word I can think of right now xD)

This would imply also:

* [ ] Modify the [Docker Compose P2P Editor port mapping](https://github.com/CodelyTV/p2p-editor/blob/master/docker-compose.yml#L6)
* [ ] Modify the [Docker Compose Signalhub port mapping](https://github.com/CodelyTV/p2p-editor/blob/master/docker-compose.yml#L12)
* [ ] Modify the [default `SIGNALHUB_URL` environment variable value](https://github.com/CodelyTV/p2p-editor/blob/master/.env.dist#L1)